src/runtime_src/xdp/profile/plugin/aie_debug/aie_debug_plugin.cpp

```diff
@@ -0,0 +1,397 @@
+/**
+ * Copyright (C) 2021 Xilinx, Inc
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"). You may
+ * not use this file except in compliance with the License. A copy of the
+ * License is located at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+#define XDP_SOURCE
+
+#define AIE_OFFSET_CORE_STATUS     0x32004
+#define AIE_OFFSET_PROGRAM_COUNTER 0x30280
+
+#include "xdp/profile/plugin/vp_base/info.h"
+#include "xdp/profile/plugin/aie_debug/aie_debug_plugin.h"
+#include "xdp/profile/writer/aie_debug/aie_debug_writer.h"
+
+#include "core/common/message.h"
+#include "core/common/system.h"
+#include "core/common/time.h"
+#include "core/common/config_reader.h"
+#include "core/include/experimental/xrt-next.h"
+#include "core/edge/user/shim.h"
+#include "xdp/profile/database/database.h"
+
+#include <boost/algorithm/string.hpp>
```

Comment:
Are you using this?
I am curious about which algorithm from there is used since I have not seen any obvious use by browsing the code.